packages/block-library/src/template-part/edit/index.js

```diff
@@ -75,7 +70,10 @@ export default function TemplatePartEdit( {
 			return {
 				innerBlocks: getBlocks( clientId ),
 				isResolved: hasResolvedEntity,
-				isMissing: hasResolvedEntity && isEmpty( entityRecord ),
+				isMissing:
+					hasResolvedEntity &&
+					entityRecord &&
+					Object.keys( entityRecord ).length > 0,
```

Comment:
Unless I'm missing something, this seems to have the inverse logic.

Shouldn't this be

```js
hasResolvedEntity && 
! ( 
  entityRecord && 
  Object.keys( entityRecord ).length > 0
)
```
or
```js
hasResolvedEntity &&
(
  ! entityRecord ||
  Object.keys( entityRecord ).length === 0
)
```
?